### PR TITLE
add lattice gauge theory example

### PR DESCRIPTION
**Pull request by maolinml**
We use Bloqade to reproduce the lattice gauge theory studied in this paper https://journals.aps.org/prx/pdf/10.1103/PhysRevX.10.021041. In particular, we use the feature of local detuning to create defects, from which to study the propagation of particle and antiparticle propagation, and reproduce Fig 5 of the paper. The example has been proofread, and the format has been checked with .ci/run doc serve, so should be good to go.

### CODE REVIEW

File: examples/6.LGT/main.jl [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+
+# We first import the required packages 
+
+using Bloqade
+using PythonCall
+using StatsBase
+using Distributed
+using BitBasis
+using Plots
```
Comment (Roger-luo) [inline examples/6.LGT/main.jl:15]: This is not necessary

File: examples/6.LGT/main.jl [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+ax1.set_ylabel("Ω1/2π (MHz)")
+ax2.set_ylabel("Δ1/2π (MHz)")
+ax1.grid()
+ax2.grid()
+fig1
+
+
+# In order to simulate the gauge theory dynamics, we define the function `Get_Ryd_density` which takes in a given set of detuning and Rabi frequency $(Δ, Ω)$, and returns the final state and the Rydberg density. 
+
+function Get_Ryd_density(Δ, Ω; dt=1e-3)
```
Comment (Roger-luo) [inline examples/6.LGT/main.jl:57]: Can we follow the snake case convention? This name is also a bit confusing with the API we export (rydberg_density). A longer and more self-explainable name would be nice.

File: examples/6.LGT/main.jl [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+clocks = 0:1e-3:Δ3.duration;
+clocks = clocks[ind0: end];
+
+# Then we plot the Rydberg density as a function of time, where the two panels correspond to the cases with single and two defects respectively
+yticks = range(clocks[1], stop=clocks[end], length=10);
+yticks = [string(ytick)[1:3] for ytick in yticks];
+
+h1 = heatmap(real(transpose(D_single_defect)), legend=:none, xlabel="sites", ylabel="time (μs)", yticks=(range(1, length(clocks), length=10), yticks));
+h2 = heatmap(real(transpose(D_two_defects)), legend=:none, xlabel="sites", yticks=(range(1, length(clocks), length=10), yticks));
+l = @layout[grid(1,2) a{0.05w}]
```
Comment (Roger-luo) [inline examples/6.LGT/main.jl:208]: Maybe let's stick to matplotlib instead of Plots.jl?

**Review by Roger-luo — CHANGES_REQUESTED** [review]

**Comment by Roger-luo** [comment]
@fanglifl could you help review the content? I think you are the best person to review this content. I'll help making the example run.

**Comment by fanglifl** [comment]
> @fanglifl could you help review the content? I think you are the best person to review this content. I'll help making the example run.

sure, it is on my to-do list for this week! 

Thanks @maolinml, this example is really a great addition to the tutorial! I will let you know my comments soon.

**Review by fanglifl — APPROVED** [review]
@maolinml all look good to me! This is really a great example to add. 

Just a minor suggestion: I think there are people interested in simulating the dynamics under the pure PXP model (constrained model without the long-range interacting tails). In the current example, the dynamics is run in the subspace, while the long-range tails are still kept. 
So perhaps we could also add dynamics from pure PXP model just for educational purpose?  The building of PXP model is a little from using rydberg_h, see the last section in the subspace page: https://queracomputing.github.io/Bloqade.jl/dev/subspace/


@Roger-luo I think the wrapper function by Mao get_average_rydberg_densities could be very useful. We could perhaps consider having a similar API for general users. The input could be get_average_rydberg_densities(atoms, \omega, \delta, dt), and the output could be the time-dependent density for each site. This is essentially what we do for most of the examples:)

**Comment by maolinml** [comment]
Hi @fanglifl, thanks for the suggestion. Quick question, how to create a time-varying PXP Hamiltonian? I tried the following but it does not seem to work
```
h = 2π * Omega * SumOfX(length(atoms)) - 2π * Delta * SumOfN(length(atoms))
```

**Comment by fanglifl** [comment]
> Hi @fanglifl, thanks for the suggestion. Quick question, how to create a time-varying PXP Hamiltonian? I tried the following but it does not seem to work
> 
> ```
> h = 2π * Omega * SumOfX(length(atoms)) - 2π * Delta * SumOfN(length(atoms))
> ```

great question! The following expression should work: 
h = SumOfX(21, Ω) - SumOfN(21, Delta)

Btw, as we discussed on slack, the PXP model could be separated to a different example, so let us keep as it is in this tutorial. I think this current example is ready to be merged! Thanks agains @maolinml

**Comment by maolinml** [comment]
Thanks @fanglifl. I will have a final edit which changes the local Rabi frequency to the local detuning fields. I will push the changes in two hours.

**Review by fanglifl — APPROVED** [review]
LGTM

**Comment by Roger-luo** [comment]
> @Roger-luo I think the wrapper function by Mao get_average_rydberg_densities could be very useful. We could perhaps consider having a similar API for general users. The input could be get_average_rydberg_densities(atoms, \omega, \delta, dt), and the output could be the time-dependent density for each site. This is essentially what we do for most of the examples:)

@maolinml since you created the function, do you want to submit a separate PR for a high-level wrapper function for it? I can help you polish the implementation. I polish a few format issues. rest LGTM.

**Review by Roger-luo — APPROVED** [review]

**Pull request merged**